### PR TITLE
Hotfix 1.1.1: Fix payout history date range.

### PR DESCRIPTION
**Pull request by shamilkhan**

### CODE REVIEW

File: src/pages/Payouts/index.tsx [hunk 1]
```diff
@@ -68,12 +68,14 @@ export const Payouts = (props: PageProps) => {
               <OpenAssistantIcon page="payouts" title="Payout History" />
             </h4>
             <h2>
-              {payouts.length ? (
+              {payoutsList.length ? (
```
[the remark this answers is not in the document]
Comment (shamilkhan) [reply src/pages/Payouts/index.tsx:71]: 1. The right order is everywhere.
Screenshot 2023-06-01 at 18 20 02
Screenshot 2023-06-01 at 18 20 21

2. If you mean the last changes in original staking repo, then they use different logic with `payoutsFromDate` and `payoutsLastDate`. Check [the source](https://github.com/paritytech/polkadot-staking-dashboard/blob/main/src/pages/Payouts/index.tsx#L32).

Comment (AndreiNavoichyk) [inline src/pages/Payouts/index.tsx:71]: Why do you use `payoutsList` instead of `payouts`?
Why did order change?

We want to minimise changes in the original code. Let's understand do we really need to introduce them.

**Review by AndreiNavoichyk — CHANGES_REQUESTED** [review]

File: src/pages/Payouts/index.tsx (hunk 1 above)
Comment (shamilkhan) [reply src/pages/Payouts/index.tsx:71]: 1. `payouts` shows all data but `payoutsList` only 60 last days with no-amount values. 
Slicing of last `MAX_PAYOUT_DAYS` is here:
```ts
 // take non-zero rewards in most-recent order
  let payoutsList: AnySubscan = [
    ...payouts.concat(poolClaims).filter((p: AnySubscan) => p.amount > 0),
  ].slice(0, MAX_PAYOUT_DAYS);
  ```
2. We change the order because `payoutsList` in the codebase change the order of the list.  (_re-order rewards based on block timestamp_)
```ts
// re-order rewards based on block timestamp
  payoutsList = payoutsList.sort((a: AnySubscan, b: AnySubscan) => {
    const x = new BN(a.block_timestamp);
    const y = new BN(b.block_timestamp);
    return y.sub(x);
  });
  ```

Comment (AndreiNavoichyk) [reply src/pages/Payouts/index.tsx:71]: @shamilkhan 
We should not change original code without the reason. In your answer I did not find the reason why should we start using `payoutsList` instead of `payouts`. Please, clarify.

Comment (shamilkhan) [reply src/pages/Payouts/index.tsx:71]: Let me clarify: `payouts` refer to all the data about payments, for example, for a stash account `6S4mrsCrqWoBAYrp2PKQNh7CYcCtyEtYpx5J626Kj5vszSyy`, which has 481 payment records. `payoutsList` is a list of the last 60 non-zero values taken from `payouts`. It is sorted in reverse order to show payments from the latest one.

The main question is why we should use `payoutsList` instead of `payouts` to show the date range. Here are a couple of reasons:

1. The chart represents data for `payoutsList`, not for `payouts`. In my opinion, it's incorrect to show the date range for 481 payment records and display only 60 of them.
2. In the original repository, they use the properties `payoutsStart` and `payoutsEndDate`, which literally represent the start date and end date of `payoutsList`. You can check the codebase [here](https://github.com/paritytech/polkadot-staking-dashboard/blob/main/src/contexts/Subscan/index.tsx#L93)

Therefore, I believe that using `payouts` instead of `payoutsList` is incorrect behavior because:

1. It does not represent the chart data correctly.
2. In the latest version, they use a property that represents `payoutsList`.

Based on this reasoning, I have decided to use this approach.

Comment (AndreiNavoichyk) [reply src/pages/Payouts/index.tsx:71]: You receive `payouts` from Cere Stats. Should it be the full account history by interface that was used previously in Subscan implementation?

Besides that your logic looks correct.

**Review by AndreiNavoichyk — APPROVED** [review]

File: src/pages/Payouts/index.tsx (hunk 1 above)
Comment (shamilkhan) [reply src/pages/Payouts/index.tsx:71]: In my opinion, yes:
1. By default, data received from the Stats App is not sorted. If you fetch N rows, it does not necessarily mean that you will receive the latest N records for rewards.
2. We need to sort non-zero values. The data is presented in a `"accountName","amount"` format, so we need to use regex on the client side to extract this value and sort.

Comment (AndreiNavoichyk) [reply src/pages/Payouts/index.tsx:71]: Let me rephrase. Does Subscan implementation return the whole history of `payouts` for some account?

Comment (shamilkhan) [reply src/pages/Payouts/index.tsx:71]: Subscan has a property called `raw` which determines the number of rows returned. Currently, in the codebase, this number is set to `100`. 

```ts
 const results = await Promise.all([
        handleFetch(activeAccount, 0, API_ENDPOINTS.subscanPoolRewards),
        handleFetch(activeAccount, 1, API_ENDPOINTS.subscanPoolRewards),
      ]);
     
     const handleFetch = async (
    address: string,
    page: number,
    endpoint: string
  ): Promise<AnySubscan> => {
    const res: Response = await fetch(network.subscanEndpoint + endpoint, {
      headers: {
        'Content-Type': 'application/json',
        'X-API-Key': API_SUBSCAN_KEY,
      },
      body: JSON.stringify({
        row: 100,
        page,
        address,
      }),
      method: 'POST',
    });
    const resJson: AnySubscan = await res.json();
    return resJson;
  };
```
